### DISCUSSION TITLE
`imp` Setup

### DISCUSSION THREAD

**Discussion by mangkoran** [arkane-systems/bottle-imp#7] [Q&A] [answered]
I would like to ask a question regarding `imp` setup.

If I understand it correctly, first after installing `imp`, we should update our WSL launch command at Windows, appending `imp -s`.

However, what I am a bit confused is the `imp -i`. If we should run `imp -i` initially, where can we put the command? Should we put it in `.bashrc`?

**Comment by cerebrate** [comment, marked as answer]
Ideally, you need to run `wsl -e imp -i` once (and let it run to completion, with systemd fully started) before using any of the other imp commands.

At the moment, I'm running it manually from PowerShell since _imp_ is new enough that I don't quite trust it to work every time. :grin: In the longer run, as with `genie -i`, I'll probably put it in a script run from Task Scheduler on logon, to automatically start up WSL at that time.

**Reply by mangkoran** [reply]
Thank you for your response. Based on this, I might just stick to run `imp -i` manually like you too.

But then after I further read the README, I think I could just use `imp -s` instead without `imp -i`. I have no use of WSL/Windows interop anyway.

**Reply by cerebrate** [reply]
And that's perfectly valid, as long as you don't want its other feature of holding the WSL instance open either.
